### PR TITLE
adds minor design polish and frame refactor

### PR DESCRIPTION
**Pull request by noahamac**
Some more design polish items I implemented as I was conducting the final design spec audit. 

Style changes were made to Header, DiagramSettings, ViewOptions, DiagramHelp, MetadataPanel, and Minimap. Tickets were made for remaining items.

Notable changes:
New error state text:
image
image
Minimap simplification:
image
More compact metadata panel:
image

Also better groups component files.

### CODE REVIEW

File: src/components/DiagramFrame/FramePanels/DiagramSettings.tsx [hunk 1]
```diff
@@ -64,9 +63,9 @@ export const DiagramSettings: React.FC<DiagramSettingsProps> = ({
  }) => {
   const history = useHistory()
   return (
-    <SettingsPanel width="275px" px="medium" py="large">
+    <SettingsPanel width="275px" p="medium">
       <SpaceVertical>
-        <Heading fontSize="large">Diagram Settings</Heading>
+        <Heading fontSize="large" color={theme.colors.text4}>Diagram Settings</Heading>
```
Comment (bryans99) [inline src/components/DiagramFrame/FramePanels/DiagramSettings.tsx:68]: Is this the recommended way to define colors now?

File: src/components/DiagramFrame/FramePanels/DiagramSettings.tsx [hunk 2]
```diff
@@ -82,46 +81,45 @@ export const DiagramSettings: React.FC<DiagramSettingsProps> = ({
...
+            />
+            <Divider appearance="light" my="medium" />
+            <Label fontSize="small" style={{marginTop: "0rem"}}>Select an Explore</Label>
+            <ExploreList style={{marginTop: theme.sizes.xxxsmall, overflow: "auto"}}>
+              {exploreList && exploreList.map((explore: ExploreDropdown, index: number) => {
+                return (
+                  <ExploreListitem key={`explore-${index}`} style={{backgroundColor: getExploreListItemBackgroundColor(explore.value, currentExplore, diagramExplore)}}>
+                    <ExploreButton
+                      onClick={() => {
+                        selectionInfo.lookmlElement === "explore" || setSelectionInfo({})
```
Comment (bryans99) [inline src/components/DiagramFrame/FramePanels/DiagramSettings.tsx:101]: consider externalizing this function from the JSX. Can this functionality be moved to the ExploreButton itself?

File: src/components/DiagramFrame/FramePanels/DiagramSettings.tsx [hunk 3]
```diff
@@ -82,46 +81,45 @@ export const DiagramSettings: React.FC<DiagramSettingsProps> = ({
...
+              onChange={(value)=>{setSelectedBranch(value)}}
+              disabled={(gitBranch && gitBranch.is_production) || !diagramExplore}
+            />
+            <Divider appearance="light" my="medium" />
+            <Label fontSize="small" style={{marginTop: "0rem"}}>Select an Explore</Label>
+            <ExploreList style={{marginTop: theme.sizes.xxxsmall, overflow: "auto"}}>
+              {exploreList && exploreList.map((explore: ExploreDropdown, index: number) => {
+                return (
+                  <ExploreListitem key={`explore-${index}`} style={{backgroundColor: getExploreListItemBackgroundColor(explore.value, currentExplore, diagramExplore)}}>
+                    <ExploreButton
```
Comment (bryans99) [inline src/components/DiagramFrame/FramePanels/DiagramSettings.tsx:99]: Can ExploreButton be moved to ExploreListItem component?

File: src/d3-utils/tables.ts [hunk 4]
```diff
@@ -99,31 +99,31 @@ export function createLookmlViewElement(
   tableData.length > 1 && tableBottomCap()
 
   // Add datatype icon
-  tableRow.append("path")
+  type !== "minimap" && tableRow.append("path")
```
Comment (bryans99) [inline src/d3-utils/tables.ts:102]: Wrap all of these with
```
if (type !== 'minimap') {
 . . .
}
```

**Review by bryans99 — COMMENTED** [review]
I've made some comments. It's up to you if you want to make another set of changes to this PR or merge and spin up a new one.

In the future, try and make small iterative PRs.

File: src/components/DiagramFrame/FramePanels/DiagramSettings.tsx (hunk 1 above)
Comment (noahamac) [reply src/components/DiagramFrame/FramePanels/DiagramSettings.tsx:68]: I would prefer a styled component but tbh I thought you liked the inline prop mechanism more. I'll follow your standard.

The prop is nice as long as it's not reused, perhaps? 

https://docs.google.com/document/d/1zyAGFHYST7oX8CHs9geyj-dOIgRVJX2Kp_zMLTetXfc/edit?usp=sharing

File: src/d3-utils/tables.ts (hunk 4 above)
Comment (noahamac) [reply src/d3-utils/tables.ts:102]: Lol... Thanks

File: src/components/DiagramFrame/FramePanels/SettingsPanel/ExploreList.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  setViewPosition,
+  setMinimapUntoggled,
+  setMinimapEnabled,
+}) => {
+  const history = useHistory()
+  return (
+    <ExploreListWrapper>
+      {exploreList && exploreList.map((explore: ExploreDropdown, index: number) => {
+        return (
+          <ExploreListitem key={`explore-${index}`} style={{backgroundColor: getExploreListItemBackgroundColor(explore.value, currentExplore, diagramExplore)}}>
```
Comment (noahamac) [inline src/components/DiagramFrame/FramePanels/SettingsPanel/ExploreList.tsx:53]: I think this is another case where inline styling is OK

File: src/d3-utils/tables.ts [hunk 6]
```diff
@@ -34,7 +34,7 @@ export function createLookmlViewElement(
   .append("g")
   .attr("class", "table-"+header.view)
 
-  table.append("rect")
+  type !== "minimap" && table.append("rect")
```
Comment (noahamac) [inline src/d3-utils/tables.ts:37]: This one needs to be done before the others

File: src/d3-utils/tables.ts [hunk 7]
```diff
@@ -98,42 +98,41 @@ export function createLookmlViewElement(
...
-  .attr('x1', 0 - (DIAGRAM_FIELD_STROKE_WIDTH/2))
-  .attr('y1', TABLE_ROW_HEIGHT + 3)
-  .attr('x2', TABLE_WIDTH + (DIAGRAM_FIELD_STROKE_WIDTH/2))
-  .attr('y2', TABLE_ROW_HEIGHT + 3)
-  .attr('class', (d: DiagramField, i: number) => {
-    return (i === 0 || i === (tableData.length - 1)) || "row-divider"
-  })
-
+  // Add diagram details
+  if (type !== "minimap") {
```
Comment (noahamac) [inline src/d3-utils/tables.ts:102]: Good call on this one... I don't know how I missed it :D

**Comment by noahamac** [comment]
@bryans99 Made some changes from your comments... thanks! Refactoring out the ExploreButton functionality was a good call :+1:

File: src/components/DiagramFrame/FramePanels/DiagramSettings.tsx (hunk 3 above)
Comment (noahamac) [reply src/components/DiagramFrame/FramePanels/DiagramSettings.tsx]: Nice. yes

File: src/components/DiagramFrame/DiagramHeader.tsx [hunk 8]
```diff
@@ -48,7 +48,7 @@ export const DiagramHeader: React.FC<DiagramHeaderProps> = ({
     >
       <Space between>
         <Space gap="xsmall">
-          <Heading as="h1">{currentExplore && currentExplore.label}</Heading>
+          <Heading as="h1" px='1rg'>{currentExplore && currentExplore.label}</Heading>
```
Comment (bryans99) [inline src/components/DiagramFrame/DiagramHeader.tsx:51]: WHat does px='1rg' do?

File: src/components/DiagramFrame/FrameHelpers.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+`
+
+export const DiagramHeaderWrapper = styled(Header as any)`
+background-color: ${(props) => props.theme.colors.background};
+border-bottom: solid 1px ${(props) => props.theme.colors.ui2};
+transition: transform 500ms ease-out;
+position: relative;
+z-index: 1;
+
+&.no-explore {
```
Comment (bryans99) [inline src/components/DiagramFrame/FrameHelpers.tsx:51]: Eye candy :D

File: src/components/DiagramFrame/FrameHelpers.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  transform: translateY(-${DIAGRAM_HEADER_HEIGHT}px);
+}
+
+&.has-explore {
```
Comment (bryans99) [inline src/components/DiagramFrame/FrameHelpers.tsx:55]: Eye candy

File: src/components/DiagramFrame/FrameHelpers.tsx (hunk 9 above)
Comment (bryans99) [reply src/components/DiagramFrame/FrameHelpers.tsx:51]: I git this to work but its very subtle fading out

File: src/components/DiagramFrame/FramePanels/SettingsPanel/ExploreList.tsx (hunk 5 above)
Comment (bryans99) [reply src/components/DiagramFrame/FramePanels/SettingsPanel/ExploreList.tsx:53]: I still don't like it but I guess I can live with it

File: src/components/DiagramFrame/MetadataPanel/QueryChart.tsx [hunk 11]
```diff
@@ -146,7 +147,7 @@ export class QueryChart extends React.Component<
     } else if (this.state.response) {
       if (this.state.response.data.length === 0) {
         return (
-          <Text fontSize="small">
+          <Text fontSize="small" style={{color: theme.colors.text1}}>
```
Comment (bryans99) [inline src/components/DiagramFrame/MetadataPanel/QueryChart.tsx:150]: I guess - still dont like it

File: src/components/DiagramFrame/MetadataPanel/QueryChartButton.tsx [hunk 12]
```diff
@@ -29,10 +29,9 @@ import { theme, ButtonOutline, FlexItem, Text } from "@looker/components"
 import styled from "styled-components"
 
 const DisabledText = styled.div`
-  font-style: italic;
-  font-size: ${theme.fontSizes.small};
-  color: ${theme.colors.ui4};
-  margin-top: 0.2em;
+  font-size: ${theme.fontSizes.xsmall};
```
Comment (bryans99) [inline src/components/DiagramFrame/MetadataPanel/QueryChartButton.tsx:32]: Prefer this over use of style attribute

**Review by bryans99 — APPROVED** [review]
Approving although I'm not keen on the use of styles. We should have an exercise to clean them up. I think the biggest thing however is the move to redux. That is a lower priority now however (at least I think it is)

**Comment by noahamac** [comment]
Will make pre GA ticket for style audit / clean up. Thanks!

**Pull request merged**